service/src/main/kotlin/fi/espoo/evaka/shared/security/Action.kt

```diff
@@ -218,37 +218,37 @@ sealed interface Action {
             override fun toString(): String = "${javaClass.name}.$name"
         }
         enum class AssistanceNeedDecision(override vararg val defaultRules: ScopedActionRule<in AssistanceNeedDecisionId>) : ScopedAction<AssistanceNeedDecisionId> {
-            READ(IsCitizen(allowWeakLogin = false).guardianOfChildOfAssistanceNeedDecision()),
-            DOWNLOAD(IsCitizen(allowWeakLogin = false).guardianOfChildOfAssistanceNeedDecision()),
-            MARK_AS_READ(IsCitizen(allowWeakLogin = false).guardianOfChildOfAssistanceNeedDecision());
+            READ(IsCitizen(allowWeakLogin = false).guardianOfChildOfAssistanceNeedDecision(), IsCitizen(allowWeakLogin = false).fosterParentOfChildOfAssistanceNeedDecision()),
+            DOWNLOAD(IsCitizen(allowWeakLogin = false).guardianOfChildOfAssistanceNeedDecision(), IsCitizen(allowWeakLogin = false).fosterParentOfChildOfAssistanceNeedDecision()),
+            MARK_AS_READ(IsCitizen(allowWeakLogin = false).guardianOfChildOfAssistanceNeedDecision(), IsCitizen(allowWeakLogin = false).fosterParentOfChildOfAssistanceNeedDecision());
 
             override fun toString(): String = "${javaClass.name}.$name"
         }
         enum class Child(override vararg val defaultRules: ScopedActionRule<in ChildId>) : ScopedAction<ChildId> {
             READ(IsCitizen(allowWeakLogin = false).guardianOfChild()),
-            READ_PLACEMENT_STATUS_BY_APPLICATION_TYPE(IsCitizen(allowWeakLogin = false).guardianOfChild()),
-            READ_DUPLICATE_APPLICATIONS(IsCitizen(allowWeakLogin = false).guardianOfChild()),
-            READ_VASU_DOCUMENT_SUMMARIES(IsCitizen(allowWeakLogin = false).guardianOfChild()),
-            CREATE_ABSENCE(IsCitizen(allowWeakLogin = true).guardianOfChild()),
+            READ_PLACEMENT_STATUS_BY_APPLICATION_TYPE(IsCitizen(allowWeakLogin = false).guardianOfChild(), IsCitizen(allowWeakLogin = false).fosterParentOfChild()),
+            READ_DUPLICATE_APPLICATIONS(IsCitizen(allowWeakLogin = false).guardianOfChild(), IsCitizen(allowWeakLogin = false).fosterParentOfChild()),
+            READ_VASU_DOCUMENT_SUMMARIES(IsCitizen(allowWeakLogin = false).guardianOfChild(), IsCitizen(allowWeakLogin = false).fosterParentOfChild()),
+            CREATE_ABSENCE(IsCitizen(allowWeakLogin = true).guardianOfChild(), IsCitizen(allowWeakLogin = true).fosterParentOfChild()),
 
-            CREATE_HOLIDAY_ABSENCE(IsCitizen(allowWeakLogin = true).guardianOfChild()),
-            CREATE_RESERVATION(IsCitizen(allowWeakLogin = true).guardianOfChild()),
+            CREATE_HOLIDAY_ABSENCE(IsCitizen(allowWeakLogin = true).guardianOfChild(), IsCitizen(allowWeakLogin = true).fosterParentOfChild()),
```

Comment:
Maybe worth having a combined guardianOrFosterParentOfChild function to shorten this?

Reply:
Yeah, good idea, since we have no need to authorize only foster parents.

Reply:
Actually now that I think of it I think it's better that the rules are separate so that we don't have to create multiple variations if there is ever a need to separate them.